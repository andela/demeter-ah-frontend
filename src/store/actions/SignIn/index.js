import { axiosCall, saveToLocalStorage } from '../../../utils';

export const signInPending = () => ({
  type: 'SIGNIN_PENDING',
  payload: {
    isCompleted: false,
    isLoading: true,
    error: null
  }
});

export const signInSuccess = user => ({
  type: 'SIGNIN_SUCCESS',
  payload: {
    isLoading: false,
    user,
    isCompleted: true,
    isAuthenticated: true,
  }
});
export const signInError = error => ({
  type: 'SIGNIN_ERROR',
  payload: {
    error,
    isCompleted: true,
    isLoading: false,
  }
});
export const cleanUpAuth = () => ({
  type: 'CLEAN_UP',
  payload: {
    error: null,
    isCompleted: false
  }
});

export const signInAction = values => async (dispatch) => {
  await dispatch(signInPending());
  try {
    const result = await axiosCall({ path: '/api/v1/users/login', payload: values, method: 'post' });
    saveToLocalStorage(result.user);
    dispatch(signInSuccess(result.user));
  } catch (error) {
    /* istanbul ignore next */
    const { response } = error;
    /* istanbul ignore next */
    const message = response.data.message || response;
    /* istanbul ignore next */
    dispatch(signInError(message));
  }
};

export const socialSignInAction = ({ username, token, history }) => async (dispatch) => {
  try {
    await dispatch(signInPending());
    const result = await axiosCall({ method: 'get', path: `/api/v1/profiles/${username}` });
    saveToLocalStorage({ ...result.user, token });
    await localStorage.setItem('socialLogin', true);
    await dispatch(signInSuccess(result.user));
    history.push('/');
  } catch ({ response }) {
    /* istanbul ignore next */
    const message = response.data.message || response;
    /* istanbul ignore next */
    await dispatch(signInError(message));
  }
};
