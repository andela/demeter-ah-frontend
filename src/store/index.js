import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
