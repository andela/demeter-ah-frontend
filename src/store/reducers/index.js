import { combineReducers } from 'redux';
import auth from './Auth';
import articleListing from './articleListing';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';
import bookmarks from './bookmarks';
import bookmarkArticle from './bookmarkArticle';
import comments from './Comments';
import search from './search';
import notifications from './notifications';
import rate from './rateArticle';

const reducers = combineReducers({
  auth,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  articleListing,
  viewArticle,
  relatedArticles,
  bookmarks,
  bookmarkArticle,
  comments,
  search,
  notifications,
  rate
});

export default reducers;
