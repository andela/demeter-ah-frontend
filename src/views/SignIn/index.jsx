import './index.scss';
import React, { Fragment, useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import callToast from '../../components/Toast';
import { decryptQuery } from '../../utils';
import InputForm from '../../components/InputForm';
import Button from '../../components/Button';
import {
  signInAction,
  cleanUpAuth,
  socialSignInAction,
} from '../../store/actions/SignIn';
import '../../styles/react-toastify.css';
import brandLogo from '../../assets/images/brand-logo.png';
import goggleIcon from '../../assets/images/google.png';
import facebookIcon from '../../assets/images/facebook.png';
import twitterIcon from '../../assets/images/twitter.png';

const SignIn = ({
  cleanUp,
  onSignIn,
  isCompleted,
  history,
  error,
  location,
  socialSignIn,
  isLoading,
}) => {
  const [values, setValues] = useState({
    email: '',
    password: '',
  });

  useEffect(
    () => {
      if (isCompleted && !error) {
        callToast('Login successful', 'success');
        history && history.push('/');
      }
      if (error) {
        callToast('Invalid email or password', 'error');
      }
      if (localStorage.socialLogin === 'false' && location.search) {
        const urlParams = new URLSearchParams(location.search);
        const token = urlParams.get('token');
        const username = urlParams.get('username');
        if (!username || !token) {
          /* istanbul ignore next */
          history && history.push('/signin');
        } else if (!isCompleted) {
          try {
            socialSignIn({ username, token: decryptQuery(token), history });
          } catch (err) {
            callToast('Invalid operation', 'error');
          }
        }
      }
      /* istanbul ignore next */
      return () => {
        /* istanbul ignore next */
        cleanUp();
      };
    },
    [isCompleted, error]
  );

  const onChange = (e) => {
    e.persist();
    setValues(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  };

  const submit = async (e) => {
    e.preventDefault();
    if (values.email === '' || values.password === '') {
      callToast('No values provided', 'error');
      return;
    }
    onSignIn(values);
  };

  const handleSocial = (e) => {
    const { id } = e.target;
    localStorage.setItem('socialLogin', false);
    window.location = `${process.env.SERVER_URL}/auth/${id}`;
  };

  return (
    <Fragment>
      <div className="signin">
        <div className="bg card-wrapper flex justify-center items-center">
          <div className="card p-10 my-10 max-w-118 min-w-96 flex-col-reverse md:flex-row sm:p-4 lg:px-10 max-w-108 sm:max-w-108 md:px-4 md:py-10 md:max-w-220">
            <form
              onSubmit={submit}
              className="left-con flex flex-col flex-grow sm:px-2 md:px-5 min-w-84 sm:w-full md:w-6/12"
            >
              <h1 className="w-full pt-4 px-12 lg:px-16 left-0 sm:top-0  md:top-1.5 text-center md:text-left absolute m-0 p-0 text-2xl text-purple-650 font-semibold">
                Sign in
              </h1>
              <div className="mb-6 md:mb-0 mt-6 pr-0 flex-grow flex flex-col justify-center w-full">
                <InputForm
                  classes="inputcon-full mb-8"
                  labelname="Email"
                  name="email"
                  labelClass="block mb-1 text-sm pl-1"
                  inputType="text"
                  onChange={onChange}
                  id="email"
                  inputClass="input text-sm py-1"
                  placeholder="johndoe@example.com"
                  pattern="^[\w.]+@[\w]{2,20}.[a-z]{2,10}$"
                  title="johndoe@example.com"
                />
                <InputForm
                  classes="inputcon-full"
                  labelname="Password"
                  name="password"
                  labelClass="block mb-1 text-sm pl-1"
                  inputType="password"
                  onChange={onChange}
                  inputClass="input text-sm"
                  placeholder="Password"
                  id="password"
                  pattern="^[\w@.]{7,20}$"
                  title="Password is invalid"
                  autoComplete="off"
                />
              </div>
              <Link to="/reset-password" className="flex items-center text-purple-650 text-sm mx-2 mb-6 cursor-pointer">
                Forgot password ?
              </Link>
              <div className="w-full flex flex-col-reverse items-center mb-4 md:mb-0 md:mt-4 md:flex-row md:justify-between">
                <p className="flex items-center mx-2 my-0 text-sm">
                  {'Not a Member?'}
                  <Link
                    to="/signup"
                    className="text-purple-650 ml-2 font-semibold cursor-pointer"
                  >
                    {'Sign up'}
                  </Link>
                </p>
                <Button
                  type="submit"
                  name="Login"
                  isSubmit={isLoading}
                  classes="btn-purple w-6/12 md:mb-0 mb-4 md:w-1/3 mx-2"
                />
              </div>
            </form>
            <div className=" flex md:flex flex-grow flex-col items-center relative">
              <hr className="vline sm:hidden md:block" />
              <span className="sm:font-semibold md:font-normal">OR</span>
              <hr className="vline sm:hidden md:block" />
            </div>
            <div className="right-con mt-10 mb-4 md:my-0 flex justify-center min-w-84 sm:w-full md:w-5/12">
              <img
                src={brandLogo}
                className="mb-4 hidden md:block sm:hidden lg:block"
                alt="Authors Haven"
              />
              <Button
                onClick={handleSocial}
                id="google"
                type="button"
                name="Login"
                classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
              >
                <img src={goggleIcon} alt="Authors Haven" />
                Sign in with Google
              </Button>
              <Button
                onClick={handleSocial}
                id="facebook"
                type="button"
                name="Login"
                classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
              >
                <img
                  src={facebookIcon}
                  alt="Authors Haven"
                />
                Sign in with Facebook
              </Button>
              <Button
                onClick={handleSocial}
                id="twitter"
                type="button"
                name="Login"
                classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
              >
                <img
                  src={twitterIcon}
                  alt="Authors Haven"
                />
                Sign in with Twitter
              </Button>
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
};

const mapStateToProps = state => ({
  error: state.auth.error,
  isLoading: state.auth.isLoading,
  user: state.auth.user,
  isCompleted: state.auth.isCompleted,
  isAuthenticated: state.auth.isAuthenticated,
});

/* istanbul ignore next */
export const onSignIn = newUser => signInAction(newUser);
export const cleanUp = () => cleanUpAuth();
/* istanbul ignore next */
export const socialSignIn = userObj => socialSignInAction(userObj);

export const SignInComponent = SignIn;

export default connect(mapStateToProps, { onSignIn, cleanUp, socialSignIn })(
  SignIn
);
