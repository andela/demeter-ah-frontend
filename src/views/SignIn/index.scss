.signin{
  font-family: 'Montserrat', sans-serif;

  .card {
    width: 95%;
  }

   h1 {
    top: 1.5em;
   }

  .vline{
    @apply w-px bg-gray-150;
    height: 48%;
  }
  .left-con{
    @apply flex-grow flex flex-wrap content-around px-5
  }

  .right-con{
    @apply flex-grow flex flex-col items-center flex-wrap relative px-5;
  }

  .input{
    @apply w-full px-2
  }

  .inputcon-full{
    @apply w-full px-2 py-1;
  }

  .inputcon-half{
    @apply w-1/2 px-2 py-1
  }
}