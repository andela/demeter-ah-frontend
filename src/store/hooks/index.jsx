import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
