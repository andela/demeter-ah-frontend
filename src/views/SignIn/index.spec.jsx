import React from 'react';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import SignIn, { SignInComponent, cleanUp } from './index';
import store from '../../store';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Sign In Component', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <SignIn store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <SignIn store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(4);
    expect(component.find('input')).toHaveLength(2);
    expect(component.find('img')).toHaveLength(4);
  });

  it('should call on change props', () => {
    const event = {
      preventDefault() { },
      target: { value: 'vinay@email.com', name: 'email' }
    };
    const mockOnSignInFn = jest.fn();
    const component = mount(
      <Router>
        <SignInComponent onSignIn={mockOnSignInFn} />
      </Router>
    );
    const inputTag = component.find('InputForm').at(0);
    const submitBtn = component.find('Button').at(0);
    inputTag.simulate('change', event);
    submitBtn.simulate('submit',
      { preventDefault() { } });
    expect(mockOnSignInFn.mock.calls.length).toBe(0);
  });

  test('Should run errors if error exist and is array', () => {
    const signInStore = mockStore({
      auth:
      {
        error: [{ message: 'some error' }],
        user: {},
        isAuthenticated: false,
        isCompleted: false,
        isSubmit: false
      }
    });
    const component = mount(
      <Router>
        <SignIn store={signInStore} />
      </Router>
    );

    const label = component.find('label').at(0);
    expect(label.text()).toEqual('Email');
  });

  test('Should run errors if error exist and is string', () => {
    const SignInStore = mockStore({
      auth:
      {
        error: 'some error',
        user: {},
        isAuthenticated: false,
        isCompleted: false,
        isSubmit: false
      }
    });
    const component = mount(
      <Router>
        <SignIn store={SignInStore} />
      </Router>
    );

    const label = component.find('label').at(0);
    expect(label.text()).toEqual('Email');
  });

  test('clean up action ', () => {
    expect(cleanUp()).toEqual({ type: 'CLEAN_UP', payload: { error: null, isCompleted: false } });
  });

  test('Should call toast if is completed', () => {
    const signupStore = mockStore({
      auth:
      {
        error: null,
        user: {},
        isAuthenticated: false,
        isCompleted: true,
        isSubmit: false
      }
    });
    const component = mount(
      <Router>
        <SignIn store={signupStore} />
      </Router>
    );

    const label = component.find('label').at(0);
    expect(label.text()).toEqual('Email');
  });

  it('should test that password is match is called', () => {
    const mockOnSignUpFn = jest.fn();
    const component = mount(
      <Router>
        <SignInComponent onSignIn={mockOnSignUpFn} />
      </Router>
    );
    const inputTag = component.find('input');
    inputTag.at(0).simulate('change', {
      target:
        { name: 'email', value: 'frank@gmail.com' }
    });
    inputTag.at(1).simulate('change', {
      target:
        { name: 'password', value: '123456789' }
    });
    inputTag.at(1).simulate('change', {
      target:
        { name: 'password', value: '123456789' }
    });
    const submitBtn = component.find('button').at(0);
    submitBtn.simulate('submit',
      { preventDefault() { } });
    expect(mockOnSignUpFn.mock.calls.length).toBe(1);
  });


  it('should singin with social login', () => {
    const component = mount(
      <Router>
        <SignInComponent />
      </Router>
    );
    const buttonTag = component.find('button');
    buttonTag.at(1).simulate('click', {
      target:
      { id: 'google' }
    });
    expect(localStorage.socialLogin).toEqual('false');
  });

  it('should authenticate user signin with social login', () => {
    const location = {
      search: '?token=eyJhbGciOiJIUzI&username=kel'
    };
    const callSocialSignIn = jest.fn(() => {});
    mount(
      <Router>
        <SignInComponent location={location} socialSignIn={callSocialSignIn} />
      </Router>
    );
    expect(location.search).toEqual('?token=eyJhbGciOiJIUzI&username=kel');
  });

  it('should redirect to signin if username and token does not exist', () => {
    const location = {
      search: ''
    };
    const history = {
      push: e => e
    };
    const callSocialSignIn = jest.fn(() => {});
    mount(
      <Router>
        <SignInComponent location={location} socialSignIn={callSocialSignIn} history={history} />
      </Router>
    );
    expect(location.search).toEqual('');
  });
});
