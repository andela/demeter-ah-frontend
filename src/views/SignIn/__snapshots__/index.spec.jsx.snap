// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Component Should render along with children componenet 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(SignIn)
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <SignIn
        cleanUp={[Function]}
        error={null}
        isAuthenticated={false}
        isCompleted={false}
        isLoading={false}
        onSignIn={[Function]}
        socialSignIn={[Function]}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        user={Object {}}
      >
        <div
          className="signin"
        >
          <div
            className="bg card-wrapper flex justify-center items-center"
          >
            <div
              className="card p-10 my-10 max-w-118 min-w-96 flex-col-reverse md:flex-row sm:p-4 lg:px-10 max-w-108 sm:max-w-108 md:px-4 md:py-10 md:max-w-220"
            >
              <form
                className="left-con flex flex-col flex-grow sm:px-2 md:px-5 min-w-84 sm:w-full md:w-6/12"
                onSubmit={[Function]}
              >
                <h1
                  className="w-full pt-4 px-12 lg:px-16 left-0 sm:top-0  md:top-1.5 text-center md:text-left absolute m-0 p-0 text-2xl text-purple-650 font-semibold"
                >
                  Sign in
                </h1>
                <div
                  className="mb-6 md:mb-0 mt-6 pr-0 flex-grow flex flex-col justify-center w-full"
                >
                  <InputForm
                    classes="inputcon-full mb-8"
                    id="email"
                    inputClass="input text-sm py-1"
                    inputType="text"
                    labelClass="block mb-1 text-sm pl-1"
                    labelname="Email"
                    name="email"
                    onChange={[Function]}
                    pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                    placeholder="johndoe@example.com"
                    title="johndoe@example.com"
                  >
                    <div
                      className="inputcon-full mb-8"
                    >
                      <Label
                        classes="block mb-1 text-sm pl-1"
                        labelname="Email"
                        name="email"
                      >
                        <label
                          className="block mb-1 text-sm pl-1"
                          htmlFor="email"
                        >
                          Email
                        </label>
                      </Label>
                      <Input
                        classes="input text-sm py-1"
                        id="email"
                        inputType="text"
                        name="email"
                        onChange={[Function]}
                        pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                        placeholder="johndoe@example.com"
                        title="johndoe@example.com"
                      >
                        <input
                          className="forminput rounded py-2 bg-purple-150 input text-sm py-1"
                          id="email"
                          name="email"
                          onChange={[Function]}
                          pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                          placeholder="johndoe@example.com"
                          required={true}
                          title="johndoe@example.com"
                          type="text"
                        />
                      </Input>
                    </div>
                  </InputForm>
                  <InputForm
                    classes="inputcon-full"
                    id="password"
                    inputClass="input text-sm"
                    inputType="password"
                    labelClass="block mb-1 text-sm pl-1"
                    labelname="Password"
                    name="password"
                    onChange={[Function]}
                    pattern="^[\\\\w@.]{7,20}$"
                    placeholder="Password"
                    title="Password is invalid"
                  >
                    <div
                      className="inputcon-full"
                    >
                      <Label
                        classes="block mb-1 text-sm pl-1"
                        labelname="Password"
                        name="password"
                      >
                        <label
                          className="block mb-1 text-sm pl-1"
                          htmlFor="password"
                        >
                          Password
                        </label>
                      </Label>
                      <Input
                        classes="input text-sm"
                        id="password"
                        inputType="password"
                        name="password"
                        onChange={[Function]}
                        pattern="^[\\\\w@.]{7,20}$"
                        placeholder="Password"
                        title="Password is invalid"
                      >
                        <input
                          className="forminput rounded py-2 bg-purple-150 input text-sm"
                          id="password"
                          name="password"
                          onChange={[Function]}
                          pattern="^[\\\\w@.]{7,20}$"
                          placeholder="Password"
                          required={true}
                          title="Password is invalid"
                          type="password"
                        />
                      </Input>
                    </div>
                  </InputForm>
                </div>
                <Link
                  className="flex items-center text-purple-650 text-sm mx-2 mb-6 cursor-pointer"
                  to="/reset-password"
                >
                  <a
                    className="flex items-center text-purple-650 text-sm mx-2 mb-6 cursor-pointer"
                    href="/reset-password"
                    onClick={[Function]}
                  >
                    Forgot password ?
                  </a>
                </Link>
                <div
                  className="w-full flex flex-col-reverse items-center mb-4 md:mb-0 md:mt-4 md:flex-row md:justify-between"
                >
                  <p
                    className="flex items-center mx-2 my-0 text-sm"
                  >
                    Not a Member?
                    <Link
                      className="text-purple-650 ml-2 font-semibold cursor-pointer"
                      to="/signup"
                    >
                      <a
                        className="text-purple-650 ml-2 font-semibold cursor-pointer"
                        href="/signup"
                        onClick={[Function]}
                      >
                        Sign up
                      </a>
                    </Link>
                  </p>
                  <Button
                    classes="btn-purple w-6/12 md:mb-0 mb-4 md:w-1/3 mx-2"
                    isSubmit={false}
                    name="Login"
                    type="submit"
                  >
                    <button
                      className="btn btn-purple w-6/12 md:mb-0 mb-4 md:w-1/3 mx-2"
                      disabled={false}
                      type="submit"
                    >
                      Login
                    </button>
                  </Button>
                </div>
              </form>
              <div
                className=" flex md:flex flex-grow flex-col items-center relative"
              >
                <hr
                  className="vline sm:hidden md:block"
                />
                <span
                  className="sm:font-semibold md:font-normal"
                >
                  OR
                </span>
                <hr
                  className="vline sm:hidden md:block"
                />
              </div>
              <div
                className="right-con mt-10 mb-4 md:my-0 flex justify-center min-w-84 sm:w-full md:w-5/12"
              >
                <img
                  alt="Authors Haven"
                  className="mb-4 hidden md:block sm:hidden lg:block"
                  src="test-file-stub"
                />
                <Button
                  classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                  id="google"
                  name="Login"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="btn w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    disabled={false}
                    id="google"
                    onClick={[Function]}
                    type="button"
                  >
                    <img
                      alt="Authors Haven"
                      src="test-file-stub"
                    />
                    Sign in with Google
                  </button>
                </Button>
                <Button
                  classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                  id="facebook"
                  name="Login"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="btn w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    disabled={false}
                    id="facebook"
                    onClick={[Function]}
                    type="button"
                  >
                    <img
                      alt="Authors Haven"
                      src="test-file-stub"
                    />
                    Sign in with Facebook
                  </button>
                </Button>
                <Button
                  classes="w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                  id="twitter"
                  name="Login"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="btn w-64 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    disabled={false}
                    id="twitter"
                    onClick={[Function]}
                    type="button"
                  >
                    <img
                      alt="Authors Haven"
                      src="test-file-stub"
                    />
                    Sign in with Twitter
                  </button>
                </Button>
              </div>
            </div>
          </div>
        </div>
      </SignIn>
    </Connect(SignIn)>
  </Router>
</BrowserRouter>
`;

exports[`Sign In Component Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(SignIn)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
